apps/src/code-studio/initApp/project.js

```diff
@@ -170,6 +177,19 @@ var projects = module.exports = {
     return url + fragment + queryString;
   },
 
+  getShareUrl() {
+    if (this.isWebLab()) {
```

Comment:
Having to check what app this is in shared code could be considered code smell, though it might make sense in this situation.

Reply:
I might at least add a function comment for `getShareUrl` describing the differences between Weblab and other apps.

Reply:
:roll_eyes: Have you seen [these other parts of project.js](https://github.com/code-dot-org/code-dot-org/blob/7ccb342e2c2457668a7be8f89c5bde970f60764e/apps/src/code-studio/initApp/project.js#L510-L626)?  If we want to fix this code smell, let's introduce some polymorphism into `dashboard.project`  Until then I don't feel bad about using this file's existing knowledge of _all_ of our project types.

Good point about the function comment.  This new function is public API, I'll add documentation.

Reply:
That comment is very fair.